readthedocs/config/config.py

```diff
@@ -128,14 +129,43 @@ class InvalidConfig(ConfigError):
 
     """Error for a specific key validation."""
 
-    message_template = 'Invalid "{key}": {error}'
+    # Define the default message to show on ``InvalidConfig``
+    default_message_template = 'Invalid configuration option "{key}"'
+
+    # Create customized message for based on each particular ``key``
+    message_templates = collections.defaultdict(lambda: "{default_message}: {error}")
```

Comment:
I feel like I don't fully understand what the `default_message` and `error` are here. Is a default_message used when a custom message isn't? But it seems like the custom templates below also include the `{default_message}`, but not `error`?

Reply:
Right. The "error" comes from who calls/raises this exception and the "default message template" is always used.

I don't like the pattern because it's hard to know the final need since it's build dynamically, but that's how the config error messages were implemented. I'd like to eventually refactor it to use the common pattern we are using for "BuildUserError" and others, where the final need the user gets it's just a static string.